Applications/Spire/Source/KeyBindings/TagNameBox.cpp

```diff
@@ -0,0 +1,196 @@
+#include "Spire/KeyBindings/TagNameBox.hpp"
+#include "Spire/KeyBindings/OrderFieldAdditionalTag.hpp"
+#include "Spire/Spire/ListIndexValueModel.hpp"
+#include "Spire/Spire/LocalValueModel.hpp"
+#include "Spire/Styles/Stylist.hpp"
+#include "Spire/Ui/DropDownBox.hpp"
+#include "Spire/Ui/Layouts.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  struct AdditionalTagNameListModel : ListModel<QString> {
```

Comment:
Instead of always repeating and copying/pasting code over and over again, try to come up with general solutions that can be composed together. There is too much boilerplate and repetition.

Reply:
I came up with `TransformListModel` class to preform the conversion from the source type in a list to another type, which functions like `TransformValueModel`.

[API](https://gist.github.com/shaobinspire/f2cc40742d806ec6029113e371e30812)

Reply:
The API looks good. Focus on the other issues with this task and then this can be added as a new task in Asana.